rosidl_generator_c/test/test_interfaces.c

```diff
@@ -896,3 +904,325 @@ int test_submessages(void)
 
   return 0;
 }
+
+/**
+ * Test message with bounded array nested types
+ */
+int test_bounded_array_nested(void)
+{
+  bool res;
+  size_t size = 4;
```

Comment:
This may be my ignorance of how this section of code works, but why are we setting the size to 4 and then only using the first two elements of the array?